gateway-discovery-cm/src/main/java/org/apache/knox/gateway/topology/discovery/cm/monitor/PollingConfigurationAnalyzer.java

```diff
@@ -360,19 +370,43 @@ private DiscoveryApiClient getApiClient(final ServiceDiscoveryConfig discoveryCo
     // Record the new event query timestamp for this address/cluster
     setEventQueryTimestamp(address, clusterName, Instant.now());
 
-    // Query the event log from CM for service/cluster restart events
-    List<ApiEvent> events = queryRestartEvents(getApiClient(configCache.getDiscoveryConfig(address, clusterName)),
+    // Query the event log from CM for service/cluster start events
+    List<ApiEvent> events = queryEvents(getApiClient(configCache.getDiscoveryConfig(address, clusterName)),
                                                clusterName,
                                                lastTimestamp);
     for (ApiEvent event : events) {
-      restartEvents.add(new RestartEvent(event));
+      if(isRelevantEvent(event)) {
+        relevantEvents.add(new StartEvent(event));
+      }
+    }
+
+    return relevantEvents;
+  }
+
+  @SuppressWarnings("unchecked")
+  private boolean isRelevantEvent(ApiEvent event) {
+    boolean rc = false;
+    String command = null;
+    String status = null;
+    List<ApiEventAttribute> attributes = event.getAttributes();
+    Map<String,Object> map = getAttributeMap(attributes);
+    command = (String) ((List<String>) map.get(COMMAND)).get(0);
+    status = (String) ((List<String>) map.get(COMMAND_STATUS)).get(0);
+    if (START_COMMAND.equals(command) || RESTART_COMMAND.equals(command) &&
+        SUCCEEDED_STATUS.equals(status) || STARTED_STATUS.equals(status)) {
```

Comment:
Does this logic need to be grouped somehow? Right now this is mixing && and ||?

Reply:
I could see this being grouped as
`(START_COMMAND.equals(command) && STARTED_STATUS.equals(status)) || (RESTART_COMMAND.equals(command) && SUCCEEDED_STATUS.equals(status))`

Reply:
IIRC, && does have a higher precedence than ||, which could mean the default grouping is `START_COMMAND.equals(command) || (RESTART_COMMAND.equals(command) &&
        SUCCEEDED_STATUS.equals(status)) || STARTED_STATUS.equals(status)`, which probably still works in this case

Reply:
It also may pick up _any_ event type that has a status value of `STARTED_STATUS`

Reply:
Not at this point. They won't be mixed up between the two and even if they were they both mean success.